src/Lykke.AlgoStore.CSharp.AlgoTemplate.Services/Functions/ADX/DirectionalMovementIndexMinusFunction.cs

```diff
@@ -0,0 +1,167 @@
+﻿using Lykke.AlgoStore.CSharp.Algo.Core.Candles;
+using Lykke.AlgoStore.CSharp.Algo.Core.Functions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Lykke.AlgoStore.CSharp.AlgoTemplate.Services.Functions.ADX
+{
+    public class DirectionalMovementIndexMinusFunction : IFunction
+    {
+        private readonly int _period;
+        private readonly bool _isAverageTrueRangeSet = false;
+
+        private int _samples { get; set; }
+
+        private DMIParameters _functionParams = new DMIParameters();
+        public FunctionParamsBase FunctionParameters => _functionParams;
+
+        private double? DirectionalMovementIndexMinus { get; set; }
+        private double? SmoothedDirectionalMovementMinus { get; set; }
+        private double? PreviousSmoothedDirectionalMovementMinus { get; set; }
+
+        private Queue<double> DirectionalMovementMinuses { get; set; }
+        private double DirectionalMovementMinus { get; set; }
+
+        /// <summary>
+        /// ATR
+        /// </summary>
+        public double? AverageTrueRange { get; set; }
+        public double? PreviousAverageTrueRange { get; set; }
+
+        private Candle _previousInput;
+        private ATRFunction ATRFunction { get; set; }
+
+        public DirectionalMovementIndexMinusFunction(DMIParameters dmiParameters)
+        {
+            _period = dmiParameters.Period;
+            _functionParams = dmiParameters;
+            DirectionalMovementMinuses = _functionParams.Period == 0 ? new Queue<double>() : new Queue<double>(_functionParams.Period);
+            _isAverageTrueRangeSet = _functionParams.IsAverageTrueRangeSet;
+            AverageTrueRange = 0.0d;
+            ATRFunction = new ATRFunction(new AtrParameters() { Period = _period });
+            _samples = 0;
+        }
+
+        public double? WarmUp(IList<Candle> values)
+        {
+            if (values == null)
+                throw new ArgumentException();
+
+            foreach (var value in values)
+            {
+                _samples++;
+
+                if (!_isAverageTrueRangeSet)
+                {
+                    AverageTrueRange = ATRFunction.AddNewValue(value);
+                }
+
+                DirectionalMovementMinus = ComputeNegativeDirectionalMovement(value);
+
+                if (values.IndexOf(value) > 0 && !IsReady)
+                {
+                    DirectionalMovementMinuses.Enqueue(DirectionalMovementMinus);
+                }
+
+                SmoothedDirectionalMovementMinus = ComputeSmoothedDirectionalMovementMinus();
+                DirectionalMovementIndexMinus = ComputeNegativeDirectionalIndex();
+
+                _previousInput = value;
+            }
+
+            return DirectionalMovementIndexMinus;
+        }
+
+        public bool IsReady
+        {
+            get { return _samples > _period + 1; }
+        }
```

Comment:
Can move it to top and use expression-bodied property "public bool IsReady => _samples > _period + 1;"